lib/puppet/type/resources.rb

```diff
@@ -145,14 +145,25 @@ def user_check(resource)
 
     return false if system_users.include?(resource[:name])
     return false if unless_uids && unless_uids.include?(current_uid)
-
-    current_uid > self[:unless_system_user]
+    if current_uid.is_a?(String)
+      # Windows user; is a system user if any regex matches.
+      not windows_system_sid_regexes.any? { |regex| current_uid =~ regex }
```

Comment:
nitpick:
should be `windows_system_sid_regexes.none? { |regex| current_uid =~ regex }`

Reply:
https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-none-3F

Reply:
To me that makes it easier to read as a sentance: "in none of the windows_system_sid_regexes does the regex match current_uid"